src/fragments/lib/storage/ios/configureaccess/50_customization.mdx

```diff
@@ -0,0 +1,145 @@
+                    completion(.success("myProtectedPrefix/" + identityId + "/"))
+                } else if accessLevel == .private {
+                    completion(.success("myPrivatePrefix/" + identityId + "/"))
+                }
+            }
+        }
+    }
+
+    public func getIdentityId(completion: @escaping (Result<String, StorageError>) -> Void) {
+        Amplify.Auth.fetchAuthSession { result in
+            do {
+                let session = try result.get()
+                if let identityProvider = session as? AuthCognitoIdentityProvider {
+                    let identityId = try identityProvider.getIdentityId().get()
+                    completion(.success(identityId))
+                }
+            } catch {
+                completion(.failure(StorageError.authError("Failed to retrieve auth session", "", error)))
+            }
+        }
+    }
+}
+```
+
+Then configure the storage plugin with the resolver.
+
+```swift
+let storagePlugin = AWSS3StoragePlugin(configuration: .prefixResolver(MyDeveloperDefinedPrefixResolver()))
+Amplify.add(storagePlugin)
+```
+
+Add the IAM policy that corresponds with the prefixes defined above to enable read, write and delete operation for all the objects under path *myPublicPrefix/*:
+
+```xml
+"Statement": [
+    {
+        "Effect": "Allow",
+        "Action": [
+            "s3:GetObject",
+            "s3:PutObject",
+            "s3:DeleteObject"
+        ],
+        "Resource": [
+            "arn:aws:s3:::your-s3-bucket/myPublicPrefix/*",
+        ]
+    }
+]
+```
+
+If you want to have custom *private* path prefix like *myPrivatePrefix/*, you need to add it into your IAM policy:
+```xml
+"Statement": [
+    {
+        "Effect": "Allow",
+        "Action": [
+            "s3:GetObject",
+            "s3:PutObject",
+            "s3:DeleteObject"
+        ],
+        "Resource": [
+            "arn:aws:s3:::your-s3-bucket/myPrivatePrefix/${cognito-identity.amazonaws.com:sub}/*"
+        ]
+    }
+]
+```
+This ensures only the authenticated users has the access to the objects under the path. 
+
+#### Passthrough PrefixResolver
+
+If you would like no prefix resolution logic, such as performing S3 requests at the root of the bucket, create a prefix resolver that returns an empty string:
+
+```swift
+public func resolvePrefix(for accessLevel: StorageAccessLevel,
+                            targetIdentityId: String?) -> Result<String, StorageError> {
+    return .success("")
+}
+```
+
+#### Client validation
+
+You can also perform validation based on the access controls you have defined. For example, if you have defined Guests with no access then you can fail the request early by checking if the user is not signed in:
+
+<amplify-callout>
+
+This will only stop your app from making a call with an unauthenticated user. You must also set up your IAM policies to protect your resources. See [CLI Storage](~/cli/storage/import.md) for more details. 
```

Comment:
We have migrated from `md` to `mdx`, this can be linked with `[CLI Storage](/cli/storage/import)`

Reply:
```suggestion
This will only stop your app from making a call with an unauthenticated user. You must also set up your IAM policies to protect your resources. See [CLI Storage](/cli/storage/import) for more details. 
```